employees/tests/test_views.py

```diff
@@ -939,3 +838,157 @@ def _get_total_hours_per_project_and_percentage_from_month(self):
                     }
                 )
         return dict(sorted(hours_and_percentage_per_project.items()))
+
+
+class CreateReportViewTests(TestCase):
+    def setUp(self):
+        self.user = UserFactory()
+        self.client.force_login(self.user)
+        self.task_activity = TaskActivityTypeFactory(is_default=True)
+        self.report = ReportFactory(
+            author=self.user,
+            date=datetime.datetime.now().date(),
+            task_activities=self.task_activity,
+            work_hours=datetime.timedelta(hours=8),
+        )
+        self.report.project.members.add(self.user)
+        self.url = reverse(
+            "create-report",
+            kwargs={"year": datetime.datetime.now().date().year, "month": datetime.datetime.now().date().month},
+        )
+        self.success_url = reverse(
+            "custom-report-list",
+            kwargs={"year": datetime.datetime.now().date().year, "month": datetime.datetime.now().date().month},
+        )
+        self.data = {
+            "date": datetime.datetime.now().date(),
+            "description": "Some description",
+            "project": self.report.project.pk,
+            "author": self.user.pk,
+            "work_hours": "8:00",
+            "task_activities": self.task_activity.pk,
+        }
+        self.invalid_data = {
+            "date": datetime.datetime.now().date(),
+            "description": "",
+            "project": self.report.project.pk,
+            "author": self.user.pk,
+            "work_hours": "8:00",
+            "task_activities": self.task_activity.pk,
+        }
+
+    def test_create_report_view_should_add_new_report_on_post_when_form_is_valid(self):
+        response = self.client.post(self.url, self.data)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(Report.objects.all().count(), 2)
+        self.assertEqual(response.url, self.success_url)
+
+    def test_create_report_view_should_not_redirect_and_create_report_on_submit_when_form_is_invalid(self):
+        response = self.client.post(self.url, self.invalid_data)
+        self.assertEqual(Report.objects.all().count(), 1)
+        self.assertEqual(response.status_code, 200)
```

Comment:
That is so strange :D It returns 200 when form is invalid and 302 when form is correct? It should return 400 when form is invalid and 201 if object is created

Reply:
We’ve discussed it. I believe that this is proper behavior in this case.